### DISCUSSION TITLE
Best way to achieve a questions/answer bot

### DISCUSSION THREAD

**Discussion by jbaumgarten** [Rapptz/discord.py#9000] [Q&A] [answered]
Guys,

I try to dev a bot which handle many questions/answers which should work like:
- question
- client answer
- new question
- ...

Is there a way to achieve this in a command or should I parse on_message events?

**Comment by CaedenPH** [comment]
> Guys,
> 
> 
> 
> I try to dev a bot which handle many questions/answers which should work like:
> 
> - question
> 
> - client answer
> 
> - new question
> 
> - ...
> 
> 
> 
> Is there a way to achieve this in a command or should I parse on_message events?

You can use the `wait_for`
method.


For example
```py
message = await bot.wait_for("message")
print(message)
```

More documentation here:
https://discordpy.readthedocs.io/en/stable/api.html?highlight=wait_for#discord.Client.wait_for

**Comment by CaedenPH** [comment, marked as answer]
Another example in a command
```py
@bot.command()
async def game(ctx):
    await ctx.send("What is 2+3?")
    response = await bot.wait_for("message", check=lambda m: m.author == ctx.author and m.channel == ctx.channel)
    if response.content == "5":
        return await ctx.send("Correct!")
```
The `check` ensures that the wait for only triggers it the message was sent in the same channel by the same user

**Reply by jbaumgarten** [reply]
Thx man :)

**Reply by CaedenPH** [reply]
Np :)
